web/impact/impact/utils.py

```diff
@@ -45,6 +47,13 @@ def get_profile(user):
         return None
 
 
+def model_name_case(view, model):
+    related_model = view.kwargs.get('related_model')
```

Comment:
it seems that you are only using the 'view' argument to get the 'related model' keyword argument. having the 'view' as an argument forces it to do some odd invocations like passing in 'self' for viewsets that use it. Instead make this argument the 'related_model' and pass that in directly from wherever it gets used.

Reply:
@yotammanor @jonkiparsky @pbx I am unable to pass in related_model with the current configuration.  It's taking the first parameter put in as the instance of GeneralViewSet.  So, I'm not sure what Shanks had in mind here to remove the view instance from this helper method...  Any ideas?

Reply:
@knittingarch it looks like @shanbady is suggesting something like the following:

```
def model_name_case(related_model):
    if related_model:
        return model + '_' + related_model
    return model
```

And then extract the 'related_model' field at the place where you call it. Does this not work?

Reply:
@jonkiparsky Yes, that is also how I interpreted his comment.  Unfortunately, this doesn't work as "related_model" ends up being essentially a renamed "view" parameter.  I don't see how to get around that weirdness.